### ISSUE TITLE
Feature Request: EC2 ImageBuilder Component

### ISSUE THREAD

**Issue by breathingdust** [hashicorp/terraform-provider-aws#13315]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In December 2019 AWS announced the availability of EC2 ImageBuilder, a service that makes it easier and faster to build and maintain secure images. Image Builder simplifies the creation, patching, testing, distribution, and sharing of Linux or Windows Server images. 

### New Resource

* aws_imagebuilder_component

### Potential Terraform Configuration

```hcl
resource "aws_imagebuilder_component" "example" {
  arn                = ""
  client_token       = ""
  change_description = ""
  data               = ""
  description        = ""
  encrypted          = true 
  kms_key_id         = ""
  name               = ""
  owner              = "" 
  platform           = ""
  tags               = {}
  type               = ""
  uri                = "" 
  version            = "" 
}
```

### References

Announcement: https://aws.amazon.com/about-aws/whats-new/2019/12/introducing-ec2-image-builder/
API Reference: https://docs.aws.amazon.com/imagebuilder/latest/APIReference/Welcome.html
SDK Reference: https://docs.aws.amazon.com/sdk-for-go/api/service/imagebuilder/
* #11084

**Comment by sppwf** [comment]
Hi, it will be nice to have this

**Comment by iancward** [comment]
Is this a dupe of #11084? That one appears to encompass the full service.

**Comment by iancward** [comment]
Might be resolved by #13485.

**Comment by bflad** [comment]
A new `aws_imagebuilder_component` resource has been merged and will release with version 3.16.0 of the Terraform AWS Provider, later this week. Thank you to @Dogers for the initial implementation. 👍
